### DISCUSSION TITLE
ImageMagick instantly killed

### DISCUSSION THREAD

**Discussion by LeonardFlatlooker** [ImageMagick/ImageMagick#6693] [Contribute]
### ImageMagick version

7.1.0-16

### Operating system

MacOS

### Operating system, version and so on

Mac OS X (Big Sur) on arm64

### Description

ImageMagick gets instantly killed when I try to launch `convert --version` or `identify --version`.
When I look at the OS messages *it seems* the program has been killed by the system because of a wrong code signature.
I have very little knowledge of how this works, but from what I understood the code signing mechanism has change on arm64 architecture on M1 macs.

Note: the problem doesn't seem to come from the brew formula to install the binary because I have tried with previous versions [such as this one](https://github.com/Homebrew/homebrew-core/blob/db3179cf0212d38d933d593b5c93aa77d68577b0/Formula/imagemagick.rb). I also tried to build imagemagick locally but it didn't change anything

### Steps to Reproduce

```bash
brew install imagemagick
convert --version
# or
identify --version
```

```bash
brew reinstall -s imagemagick
convert --version
# or
identify --version
```

### Images

![image](https://user-images.githubusercontent.com/58165523/144217292-746b7829-fbff-48e1-a83d-87d2c7883ad4.png)

**Comment by dlemstra** [comment]
I think you will need to ask this question somewhere else. We don't maintain the binaries that get installed on your system with `brew`.

**Comment by LeonardFlatlooker** [comment]
It seems `brew` is getting the binaries from ImageMagick's website ([https://www.imagemagick.org/download/releases/ImageMagick-7.1.0-16.tar.xz](https://www.imagemagick.org/download/releases/ImageMagick-7.1.0-16.tar.xz)).
Besides I also have this problem when I build ImageMagick locally.

**Comment by dlemstra** [comment]
We don't have the hardware to try to reproduce this. We cannot fix this ourselves so we need to wait for a patch from the community to resolve this.
